### ISSUE TITLE
How do I get the max depth of the entire tree/subtree?

### ISSUE THREAD

**Issue by marcamillion** [stefankroes/ancestry#219]
I would love to be able to do `node.subtree.max_depth` or just `node.max_depth` and both will check the maximum amount of levels on the tree and tell me how many levels there are in the respective tree.

In the first example - `node.subtree.max_depth` would use `node` as the `root` (i.e. `depth = 0`) and `node.max_depth` would look for the root node on that tree and use that as the starting point and go from there.

What is the best way to approach this, given that this isn't built-in to the current version of the gem?

Basically what I am trying to do is to retrieve all the relatives of a node in a tree. i.e. all the children, parents, grandparents, etc. If I have some bound as to how large the tree is, it makes life easier than just an open-ended calculation.

**Comment by vanderhoorn** [comment]
Well, you could count the number of slashes in the ```ancestry``` column. Maybe PostgreSQL has a way of counting substrings in strings and then maxing it.

**Comment by stefankroes** [comment]
Turn on depth caching and use the Rails maximum method (passing :depth) on the subtree (which returns a scope).

**Comment by kbrock** [comment]
proposed solution from ~2 years ago.
Please let us know if this does not work for you.
